Tgr main

danamansana opened the pull request:
- Update ruby version to 3.3.3
- Update packages with critical dependencies
- Switch to new CI/CD strategy (add config files, `.github/workflows/test-and-deploy.yml`, `provisioning` directory for terraform, remove `.travis` and associated scripts)

danamansana committed:
Update gems, ruby version, and deployment pattern

danamansana committed:
Add new env variable config and remove travis

danamansana committed:
Remove travis deploy script

danamansana committed:
Fix name of handler

EdwinGuzman commented:
I had a ticket for this and was in the middle of the update but no problem! https://newyorkpubliclibrary.atlassian.net/browse/TGR-103 It's similar to how I picked up the proxy request service you were originally assigned.

EdwinGuzman reviewed (changes requested):
Some updates:
- update the ruby version in each sam file from `ruby2.5` to `ruby3.3`. Otherwise, the `sam local invoke --event event.json --region us-east-1 --template sam.local.yml --profile nypl-digital-dev` command fails.
- in the README replace Travis with Github Actions.
- in the README, I think it's ok to remove the `Manual deployments` section.
- can you verify that the commands in the README still work? I'm having trouble with the `same local invoke ...` command and the `aws sqs --endpoint ...` command.

nonword commented on sam.local.yml:
```diff
@@ -11,7 +11,8 @@ Resources:
       Environment:
         Variables:
-          SQS_QUEUE_URL: AQECAHh7ea2tyZ6phZgT4B9BDKwguhlFtRC6hgt+7HbmeFsrsgAAAKcwgaQGCSqGSIb3DQEHBqCBljCBkwIBADCBjQYJKoZIhvcNAQcBMB4GCWCGSAFlAwQBLjARBAzpQKyNgJtsrpREXPoCARCAYA8SWP3gvt7ZHWECqyZwOzAdwz++tpS+63mbH96MVCek4GvRDwfG7+f3IPQYV6kf7dd6AzDL2QIMaPbOS7K2sObmhvS3rIqb2RUDQk9Qf0q78l+xCYh2UF74nF93bKL7Vg==
+          # SQS_QUEUE_URL: AQECAHh7ea2tyZ6phZgT4B9BDKwguhlFtRC6hgt+7HbmeFsrsgAAAKcwgaQGCSqGSIb3DQEHBqCBljCBkwIBADCBjQYJKoZIhvcNAQcBMB4GCWCGSAFlAwQBLjARBAzpQKyNgJtsrpREXPoCARCAYA8SWP3gvt7ZHWECqyZwOzAdwz++tpS+63mbH96MVCek4GvRDwfG7+f3IPQYV6kf7dd6AzDL2QIMaPbOS7K2sObmhvS3rIqb2RUDQk9Qf0q78l+xCYh2UF74nF93bKL7Vg==
```
Prob don't need this?

danamansana committed:
Fix some issues with sam

EdwinGuzman commented on README.md:
````diff
@@ -33,7 +33,7 @@ All config is in sam.[ENVIRONMENT].yml templates, encrypted as necessary.
 The following will invoke the lambda against the sample `event.json`
 ```
-sam local invoke --event event.json --region us-east-1 --template sam.[ENVIRONMENT].yml --profile [aws profile]
+AWS_ACCESS_KEY_ID=[your access key id] AWS_SECRET_ACCESS_KEY=[your secret access key] sam local invoke --event event.json --region us-east-1 --template sam.qa.yml --profile nypl-digital-dev
````
Now that the env vars are in the command, you don't need the `--profile nypl-digital-dev` flag.

nonword replied:
True. But I also wonder why `--profile` wasn't working. I'm able to run this branch with just `--profile`. I try to keep keys out of my command history

EdwinGuzman reviewed (approved):
Works for me locally! The rest of the updates look good to me.

EdwinGuzman commented:
Also, should this point to the `development` branch instead of `qa`?

nonword commented on .ruby-version:
```diff
@@ -1 +1 @@
-2.5.0
+3.3.3
```
Perhaps just specify `3.3`, since there's already a stable 3.3.5 and in general we want those patches

nonword commented on .github/workflows/test-and-deploy.yml:
Previous git strategy was: PRs merge to development, deploy via qa and master (production) - although oddly the default branch is also (dangerously) master.

I see that with these changes we're moving to more standard qa and production deployment branches. Should we do the following?
 - Change this PR to target `master`
 - Update the README as part of this PR to indicate this app uses a standard [Main-QA-Production workflow](https://github.com/NYPL/engineering-general/blob/main/standards/git-workflow.md#main-qa-production)
 - After merging to `master`, rename `master` to `main`
 - Merge `main` > `qa` to deploy

danamansana committed:
Add small fixes post PR:
- Initialize terraform
- Add terraform files to gitignore
- Document git workflow in README
- Change KMS and SQS initialization to work on remote
